scripts/github_api

```diff
@@ -0,0 +1,171 @@
+#!/bin/python2
+from functools import partial
+import argparse
+import json
+import os
+import re
+import requests
+import sys
+
+CLIENT_ERROR = 422
+API_BASE_URL = "https://api.github.com/"
+
+
+def request(*args, **kwargs):
+    raise Exception("The '_setup_auth' function must be called once "
+                    "before a request can be made")
+
+
+def _setup_auth(user, password):
+    """
+    Sets up basic authentication to use with HTTP requests
+
+    Args:
+        user (str): GitHub user name
+        password (str): GitHub user password or token
+    """
+    auth = (user, password)
+    global request
+    request = partial(_request, auth)
+
+
+def _request(auth, method, url, **kwargs):
+    """
+    Makes HTTP requests.
+
+    Args:
+        auth (tuple): user credentials
+        method (str): HTTP method
+        url (str): REST API endpoint URL
+
+    Returns:
+        dict: HTTP response
+    """
+    methods = {
+        'POST': requests.post
+    }
+
+    r = methods[method](url, auth=auth, **kwargs)
+    response = json.loads(r.text)
+
+    try:
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as exc:
+        if r.status_code in [CLIENT_ERROR]:
+            error_msg = response['errors'][0]['message']
+            raise Exception(error_msg)
+        raise
+
+    return response
+
+
+class PullRequest:
+    """
+    https://developer.github.com/v3/pulls/
+    """
+    def __init__(self, repo):
+        """
+        Args:
+            repo (str): git repository path. Format <repo owner>/<repo>
+        """
+        self.repo = repo
+        self._url = os.path.join(
+            API_BASE_URL, "repos/%(repo)s/pulls" % dict(repo=self.repo))
+
+    def open(self, title, body, src_branch, dst_branch):
+        """
+        Opens a GitHub pull-request
+
+        Args:
+            title (str): title text
+            body (str): body text
+            src_branch (str): destination repository. Format:
+                <repo owner>/<repo>
+            dst_branch (str): destination branch
+        """
+        payload_body = {
+            "title": title,
+            "body": body,
+            "head": src_branch,
+            "base": dst_branch,
+        }
+
+        self._set_attributes(
+            request('POST', self._url, data=json.dumps(payload_body)))
```

Comment:
I'd prefer to put request() result in an intermediate 'response' or 'http_response' variable to make code clearer, but up to you